### DISCUSSION TITLE
Why are Vue3 docs showing a different way of doing things then the quasar-cli

### DISCUSSION THREAD

**Discussion by delebash** [quasarframework/quasar#10670] [General - Components / Directives / etc] [answered]
I am knew to Vue but I could not find an answer to this after searching.

In Vue3 you create an application like so
```
import Vue from "vue";
const app = Vue.createApp({})
```

But in the quasar template  you create it this way
In App.vue

```
import { defineComponent } from 'vue';

export default defineComponent({
  name: 'App'
})
```

Then for using a component Quasar has

```
import ComponentA from './components/ComponentA.vue';
export default {
  setup() {
  components: {
    ComponentA 
  }
}}

```

However in Vue3 docs they show it like this

```
import { createApp } from 'vue';
import ComponentA from './components/ComponentA.vue';
const app = createApp({}); 

app.component('ComponentA', {
  /* ... */
});

app.mount('#app');
```

So in the quasar cli app where is

import { createApp } from 'vue'  and
app.mount('#app');

Or does quasar just do it a different way and it really does not matter?

**Comment by metalsadman** [comment, marked as answer]
quasar cli generates it, check your `.quasar` folder, the generated code are there `client-entry.js`. basic gist remains the same as for your other questions.
